src/bgen/Generator.cs

```diff
@@ -2955,7 +2955,7 @@ void GetReturnsWrappers (MethodInfo mi, MemberInformation minfo, Type declaringT
 				var suffix = string.Empty;
 				print ("{0} retvalarrtmp;", NativeHandleType);
 				cast_a = "((retvalarrtmp = ";
-				cast_b = ") == IntPtr.Zero ? null! : (";
+				cast_b = ") == IntPtr.Zero ? [] : (";
```

Comment:
While it's bad practice, it's also a breaking change to change it now 😒

Reply:
Bummer, ok, let's take a diff approach to clean that code, since that check is not really needed. The CFArray already returns a null value when it receives a IntPtr.Zero. Let's clean that in a diff PR once https://github.com/xamarin/xamarin-macios/pull/22216 lands. We can do better than what we have atm.